Tests/FluentAssertions.Specs/Events/EventAssertionSpecs.cs

```diff
@@ -540,6 +540,53 @@ public void
             act.Should().Throw<XunitException>()
                 .WithMessage("Expected*property*SomeProperty*but*OtherProperty1*OtherProperty2*");
         }
+
+        [Fact]
+        public void
+            The_number_of_property_changed_recorded_for_a_specific_property_matches_the_number_of_time_it_was_raised_specifically()
+        {
+            // Arrange
+            var subject = new EventRaisingClass();
+            using var monitor = subject.Monitor();
+            subject.RaiseEventWithSenderAndPropertyName(nameof(EventRaisingClass.SomeProperty));
+            subject.RaiseEventWithSenderAndPropertyName(nameof(EventRaisingClass.SomeProperty));
+            subject.RaiseEventWithSenderAndPropertyName(nameof(EventRaisingClass.SomeOtherProperty));
+
+            // Act
+            monitor.Should().RaisePropertyChangeFor(x => x.SomeProperty).Count().Should().Be(2);
+        }
+
+        [Fact]
+        public void
+            The_number_of_property_changed_recorded_for_a_specific_property_matches_the_number_of_time_it_was_raised_including_agnostic_property()
+        {
+            // Arrange
+            var subject = new EventRaisingClass();
+            using var monitor = subject.Monitor();
+            subject.RaiseEventWithSenderAndPropertyName(nameof(EventRaisingClass.SomeProperty));
+            subject.RaiseEventWithSenderAndPropertyName(nameof(EventRaisingClass.SomeOtherProperty));
+            subject.RaiseEventWithSenderAndPropertyName(null);
```

Comment:
🤔`INotifyPropertyChanged` does not allow a `null` property. But if remove the line from this test that passes `null`, the test fails. I didn't expect that.

Reply:
The remark section of the [documentation page](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged.propertychanged?view=net-8.0#system-componentmodel-inotifypropertychanged-propertychanged) for `INotifyPropertyChanged.PropertyChanged` mentions that the property name of the `PropertyChangedEventArgs` could be `null` or `string.Empty` to target all properties.

Reply:
Yeah, you're right. I'm at loss what I was thinking at that time. However, shouldn't the assertion then not expect 4 events?

Reply:
It makes sense for me as it is, because he's checking for `x.SomeProperty` which isn't matching for `SomeOtherProperty`, but for everything else in this case.

Reply:
The purpose of this test was to make sure all events related to a single property is being returned as expected.  3 out of 4 events are raised for this purpose ("SomeProperty", `null` and `string.Empty`).  The last event is targeting "SomeOtherProperty" and should not be returned or count in the final result.  The reason it was added to the test was to validate that the function is actually returning a filtered list of events and not the full list.  To achieve this, I added an event, an intruder, that is not supposed to be returned.